src/main/scala-2.11/markets/clearing/engines/CDAMatchingEngine.scala

```diff
@@ -83,18 +83,17 @@ class CDAMatchingEngine(askOrdering: PriceOrdering[AskOrder],
         val residualQuantity = incoming.quantity - askOrder.quantity
         val price = formPrice(incoming, askOrder)
 
-        if (residualQuantity < 0) {
-          val totalMatch = TotalMatch(askOrder, incoming, price)
-          // add residualOrder back into orderBook!
-          val residualOrder = askOrder.split(-residualQuantity)
+        if (residualQuantity < 0) {  // incoming order is smaller than existing order
+          val (filledOrder, residualOrder) = askOrder.split(-residualQuantity)
+          val totalMatch = TotalMatch(filledOrder, incoming, price)
           _askOrderBook.add(residualOrder)  // SIDE EFFECT!
           matches.enqueue(totalMatch)
         } else if (residualQuantity == 0) {  // no rationing for incoming order!
           val totalMatch = TotalMatch(askOrder, incoming, price)
           matches.enqueue(totalMatch)
         } else {  // incoming order is larger than existing order and will be rationed!
-          val partialMatch = PartialMatch(askOrder, incoming, price)
-          val residualOrder = incoming.split(residualQuantity)
+          val (filledOrder, residualOrder) = incoming.split(residualQuantity)
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Field name does not match the regular expression '^[a-z][A-Za-z0-9]*$'](https://www.codacy.com/app/davidrpugh/markets-sandbox/file/1539841661/issues/source?bid=2921361&fileBranchId=3014278#l95)